Fix missing state information in scoreboard JSON report

johnbrvc opened the pull request:
### Description of what the PR does
The incorrect **CLICS** routine was being called to generate the JSON scoreboard report.  We should be using the newest **CLICS API** Scoreboard objects (2023-06). This fix involved adding Jackson deserialization capability to several of the CLICS **API202306** classes (an empty constructor and getter methods). The **FileComparisonUtilities** had to be changed to use the correct CLICS API class names.
CI: Removed unused argument "**controller**" from `API202306.CLICSScoreboard`
CI: Fix datatype error for the "**time**" property in` API202306.CLICSScore`
CI: Attempt to populate the primary and shadow results folders to speed up EOC procedures (no longer should be necessary to track down the results folder using an explorer window).

This bug was found during NAC when the shadow (DOMjudge) noticed we didn't include state information in the scoreboard JSON.  The CLICS API Endpoint returned the correct data, but the reports did not.

This is an easy PR to review.  Many of the changes are adding _getter_ methods to some classes as well as default constructors.

### Issue which the PR addresses
Fixes #1109 

### Environment in which the PR was developed (OS,IDE, Java version, etc.)
Windows 11
java version "1.8.0_321"
Java(TM) SE Runtime Environment (build 1.8.0_321-b07)
Java HotSpot(TM) 64-Bit Server VM (build 25.321-b07, mixed mode)

### Precise steps for _testing_ the PR 

1. Load a completed contest, or at the very least, a contest that has several correct submissions.
2. From an administrator client generate the: "**Results Export File**s" report.
3. Verify that the `scoreboard.json` produced is correct (contains a correct state object instead of all empty strings)
4. From an administrator client generate the: "**Scoreboard JSON**" report and save it someplace.
5. Open the saved file and notice that the state object in the JSON contains valid information as opposed to empty strings.

To test the CI regarding populating the _Primary_ and _Shadow_ folder edit fields:
Load a completed contest, such as **NAC2025**  or a contest that has a `results` folder in the CDP base folder (same level as the `config `folder).
From an administrator client, on the **Run Contest** tab, choose the **Results Compare** tab.
Notice that the folders for _Primary_ and _PC2_ are filled in.

It should be noted that the nomenclature chosen for the results folders should read _Primary_ and _Shadow_ as opposed to _Primary_ and _PC2_.  PC2 can be the primary and some other CCS can be the shadow.  These should be fixed at some point someday.

johnbrvc committed:
i_1109 Fix missing state information in scoreboard JSON report

The incorrect CLICS routine was being called to generate the JSON scoreboard report.  We should be using the newest CLICS API Scoreboard objects (2023-06).
This fix involved adding deserialization capability to several of the CLICS API202306 classes (an empty constructor and getter methods).
The FileComparisonUtilities had to be changed to use the correct CLICS API class names.
CI: Removed unused argument "controller" from API202306.CLICSScoreboard
CI: Fix datatype error for the "time" property in API202306.CLICSScore
CI: Attempt to populate the primary and shadow results folders to speed up EOC procedures (no longer should be necessary to track down the results folder using an explorer window).

johnbrvc committed:
i1109 Removed debug statements

Debug statements were added to test the auto-fill of the primary and shadow results folder.  These have been removed.

johnbrvc committed:
i_1109 Fix JUnit compile error

The JUnit for FileComparisonUtilities failed since the return type of a score board row changed.  This has been fixed.

clevengr commented on src/edu/csus/ecs/pc2/clics/API202306/CLICSScore.java:
```diff
@@ -39,8 +49,20 @@ public CLICSScore(TeamStanding teamStanding) {
         if(num_solved > 0) {
             // Problem solution time is in minutes.
-            time = teamStanding.getLastSolved();
+            time = StringUtilities.getIntegerValue(teamStanding.getLastSolved(), 0);
```
I worry that "0" is not a good default for `getIntegerValue()` -- that would mean defaulting to "the team solved this at time zero".  (I realize this "should never happen" -- but what if it does, Tammy?)  It seems like if the "default value" were ever actually used, it's an indication of a substantial problem in the code.  It seems like it would be better to at LEAST log a warning, maybe a severe, error; maybe even throw an exception at that point...

johnbrvc replied:
I was (when I <del>wrote</del> copied this) more concerned about the `total_time `line 3 lines back.  It too defaults to 0.   `getIntegerValue` will throw an exception if the value is not a number.  If sets it to the default (0) if the string is `null `or empty.  I suppose we can change it to `Integer.parseInt(teamStanding.getLastSolved())` and have it cause an exception for all non-integer values; at least we'll have a log of what happened.  Probably the same for `total_time `?  If that is satisfactory, I'll make the change.

clevengr replied:
Well, it appears that making such a change would at least avoid choosing a "last solved time" of zero if there's an error in the system, so yes that does seem better.  Although if `Integer.parseInt()` throws an exception, the value of `time` will still have its (Java-default) value of zero.  But at least we'd have an exception to look at, so... yes.

clevengr reviewed (commented):
Reviewed `src/edu/csus/ecs/pc2/clics/API202306/CLICSScore.java`.  Left one comment/concern.

clevengr reviewed (approved):
I reviewed all 11 changed files.  I commented on one place I would suggest be updated; other than that, all the changes seem reasonable.   I built the system, loaded the NAC2025 contest into it, and verified that the `scoreboard.json` files generated by both the `ReportsExportFiles` and the `Scoreboard JSON` file reports contain valid "state" info rather than empty strings.   Everything seems to work; I approve the PR.

johnbrvc committed:
i_1109 Allow exception of bad values in scoreboard deserialization

Previously, Utilities.nullSafeToInt() and StringUtilities.getIntegerValue() were used to obtain the values of total_time and time.  Basically, those routines would assign a value of zero if the string was empty or null, and, in the former case, if a bad string to integer conversion occurred.  Now, since these are for score values, we will call Integer.parseInt() which will throw an exception if bad values are found.  All values should be good at this point and this shouldn't happen, Tammy.

JoeTerlizzi reviewed (approved):
Tested with NAC2025 contest.  scoreboard.json contains valid states for state object.  Approved.
